cdc/generation.cc

```diff
@@ -463,22 +463,44 @@ static std::optional<cdc::generation_id> get_generation_id_for(const gms::inet_a
     return gms::versioned_value::cdc_generation_id_from_string(gen_id_string);
 }
 
+static future<std::optional<cdc::topology_description>> retrieve_generation_data_v2(
```

Comment:
A test which writes to CDC-enabled table while doing recovery might be useful.

Let's do it in a follow-up